src/functions_framework/__init__.py

```diff
@@ -60,13 +66,29 @@ def __init__(
         self.data = data
 
 
+def _get_cloud_event_version(request):
+    headers = request.headers
+    # TODO(joelgerard): Should there be further inspection of the payload for versioning considerations?
+    if headers.get("Content-Type") == "application/cloudevents+json":
+            return v1.Event()
+    return None
+
+def _convert_request_to_event(request, cloud_event_def):
+    try:
+        m = marshaller.NewDefaultHTTPMarshaller()
+        data = io.StringIO(request.get_data(as_text=True))
+        return m.FromRequest(cloud_event_def, request.headers, data, json.loads)
+    except:
+        raise FunctionsFrameworkException("Content-Type header indicated a Cloud Event, "
+                                          "but it could not be parsed.")
+
+
 def _http_view_func_wrapper(function, request):
     def view_func(path):
         return function(request._get_current_object())
 
     return view_func
 
-
 def _is_binary_cloud_event(request):
```

Comment:
I believe we shouldn't need this method anymore as `m.FromRequest` should handle both binary and structured requests.

The existing code here can be removed.

Reply:
Please clarify. Seemed slightly different. Happy to adjust.